WcaOnRails/app/webpacker/components/EditSchedule/EditActivities/EditActivityModal.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect } from 'react';
+import {
+  Button,
+  Container,
+  Form,
+  Modal,
+} from 'semantic-ui-react';
+import { DateTime } from 'luxon';
+import useInputState from '../../../lib/hooks/useInputState';
+
+const commonActivityCodes = {
+  'other-registration': 'On-site registration',
+  'other-checkin': 'Check-in',
+  'other-tutorial': 'Tutorial for new competitors',
+  'other-multi': 'Cube submission for 3x3x3 Multi-Blind',
+  'other-breakfast': 'Breakfast',
+  'other-lunch': 'Lunch',
+  'other-dinner': 'Dinner',
+  'other-awards': 'Awards',
+  'other-misc': 'Other',
+};
+
+const otherActivityCodeOptions = Object.keys(commonActivityCodes).map((activityCode) => ({
+  key: activityCode,
+  text: commonActivityCodes[activityCode],
+  value: activityCode,
+}));
+
+function EditActivityModal({
+  showModal,
```

Comment:
I always think this is a function to show the modal. I'd recommend `isModalOpen`. Maybe it's just me though.